cl/search/views.py

```diff
@@ -328,6 +337,39 @@ def get_homepage_stats():
     return homepage_data
 
 
+def store_search_query(request: HttpRequest, search_results: dict) -> None:
+    """Saves an user's search query in a SearchQuery model
+
+    The `hit_cache` and `query_time_ms` fields computation depend on
+    whether the search was executed via ElasticSearch or Solr
+
+    :param request: the request object
+    :param search_results: the dict returned by `do_search` or
+        `do_es_search` functions
+    :return None
+    """
+    search_query = SearchQuery(
+        user=None if request.user.is_anonymous else request.user,
+        get_params=request.GET.urlencode(),
+    )
+
+    query_time = search_results.get("results_details", [None])[0]
+    if query_time is not None:
+        search_query.query_time_ms = ceil(query_time)
+        # We set ES `query_time` metadata with values 0 or 1 if cache is hit:
+        # 0: homepage cache; 1: micro cache
+        search_query.hit_cache = query_time in [0, 1]
```

Comment:
I found that we can mistakenly set failed queries in ES as `hit_cache = True`, since failed queries also return 0 within the `fetch_es_results` method. I think we could set the query time to `None` if the query fails and adjust the return types in that method and downstream. This way, we can ensure that query times of 0 and 1 correspond to queries that actually hit the cache.